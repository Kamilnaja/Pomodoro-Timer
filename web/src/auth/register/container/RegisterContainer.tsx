import React from "react";
import {connect} from "react-redux";
import {Registration} from "../../../../../types/interfaces";
import {RegisterComponent} from "../component/RegisterComponent";
import {sendRegisterForm} from "../../store/actions/auth.actions";
import {AuthState} from "../../store/interfaces/auth.state";
import {handleCloseModal} from "../../../shared/scripts/utils";

export interface RegisterProps {
  handleSubmit: (arg: Registration) => void;
  handleClose: () => void;
  auth: AuthState;
}

class RegisterContainer extends React.Component<RegisterProps> {
  handleSubmit = (registrationData: Registration) => {
    const { login, password, email } = registrationData;
    this.props.handleSubmit({
      login,
      password,
      email,
    });
  };

  componentDidUpdate = (prevProps: RegisterProps) => {
    handleCloseModal(prevProps.auth, this.props);
  };

  render() {
    return <RegisterComponent handleSubmit={this.handleSubmit} formState={this.props.auth} handleClose={this.props.handleClose} />;
  }
}

const mapStateToProps = (state: { auth: AuthState }) => {
  const formState = state.auth;
  return { auth: formState };
};

const mapDispatchToProps = {
  handleSubmit: sendRegisterForm,
};

export default connect(mapStateToProps, mapDispatchToProps)(RegisterContainer);
