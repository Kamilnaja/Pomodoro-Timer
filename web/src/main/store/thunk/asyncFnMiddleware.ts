export async function fetchStats(dispatch: any, getState: any) {
  const response = await client.get("/fakeApi/todos");
  dispatch({ type: "todos/todosLoaded", payload: response.todos });
}
