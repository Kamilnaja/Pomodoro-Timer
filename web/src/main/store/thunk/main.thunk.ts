import TodayStatistics from "../../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
  savePomodoro,
  savePomodoroError,
  savePomodoroSuccess,
} from "../actions/actions";
// todo - different versions for prod etc
const API_URL = "http://localhost:8080/pomodoros";

function makePostRequest() {
  return fetch(API_URL, {
    method: "POST",
  });
}

export function makeGetRequest() {
  return fetch(API_URL);
}

function handleErrors(response: any): Promise<any> {
  if (!response.ok) {
    throw Error(response.statusText);
  }
  return response.json();
}

export function savePomodoroThunk() {
  return (dispatch: Function) => {
    dispatch(savePomodoro());

    return makePostRequest()
      .then(handleErrors)
      .then(() => dispatch(savePomodoroSuccess()))
      .catch((error) => dispatch(savePomodoroError(error)));
  };
}

export function getPomodorosThunk() {
  return (dispatch: Function) => {
    dispatch(getTodayStatistics());

    return makeGetRequest()
      .then(handleErrors)
      .then((payload: TodayStatistics) => {
        console.log(payload);
        return dispatch(getTodayStatisticsSuccess(payload));
      })
      .catch((error) => dispatch(getTodayStatisticsError(error)));
  };
}
