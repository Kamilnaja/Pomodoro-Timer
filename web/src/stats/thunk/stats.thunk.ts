import { handleErrors } from "../../main/store/thunk/main.thunk";
import { initialConfig } from "../../shared/settings/initialConfig";
import TodayStatistics from "../../shared/store/interfaces/todayStatistics.interface";
import {
  getTodayStatistics,
  getTodayStatisticsError,
  getTodayStatisticsSuccess,
} from "../store/actions/stats.actions";

const API_URL = initialConfig.apiUrl;

const makeGetRequest = () => fetch(API_URL + "/pomodoros_done_today");

export const getPomodorosThunk = () => (dispatch: Function) => {
  dispatch(getTodayStatistics());

  return makeGetRequest()
    .then(handleErrors)
    .then((response) => response.json())
    .then((payload: TodayStatistics) => {
      return dispatch(getTodayStatisticsSuccess(payload));
    })
    .catch((error) => dispatch(getTodayStatisticsError(error)));
};
