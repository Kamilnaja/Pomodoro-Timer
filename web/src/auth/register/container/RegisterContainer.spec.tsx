import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import { createAuthState } from '../../testing/auth.testing.data';
import { Provider } from 'react-redux';
import { ConnectedRegisterContainer } from './RegisterContainer';

const mockStore = configureStore([thunk]);

describe('RegisterContainer', () => {
  let store;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      auth: createAuthState,
    });

    component = renderer.create(
      <Provider store={store}>
        <ConnectedRegisterContainer />
      </Provider>,
    );
  });

  it('Should match the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
