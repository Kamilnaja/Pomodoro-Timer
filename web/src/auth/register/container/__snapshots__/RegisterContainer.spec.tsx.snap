// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterContainer Should match the snapshot 1`] = `
<div
  className="register"
>
  <form
    className="register__form form"
    onSubmit={[Function]}
  >
    <div
      className="form__row"
    >
      <label
        className="form__label label"
        htmlFor="login"
      >
        User login
      </label>
      <input
        autoFocus={true}
        className="form__input input"
        id="login"
        name="login"
        placeholder="John Doe"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form__row"
    >
      <label
        className="form__label label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="form__input input"
        id="email"
        name="email"
        placeholder="johndoe@gmail.com"
        required={true}
        type="email"
      />
    </div>
    <div
      className="form__row"
    >
      <label
        className="form__label label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form_input input"
        id="password"
        name="password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="form__row"
    >
      <label
        className="form__label label"
        htmlFor="repeatedPassword"
      >
        Repeat password
      </label>
      <input
        className="form__input input"
        id="repeatedPassword"
        name="repeatedPassword"
        required={true}
        type="password"
      />
    </div>
    <button
      className="form__button"
      type="submit"
      value="WyÅ›lij"
    >
      Submit
    </button>
  </form>
</div>
`;
